src/particles/plasma/PlasmaParticleContainer.cpp

```diff
@@ -624,30 +627,88 @@ LaserIonization (const int islice,
-        amrex::ParallelFor(num_ions,
-            [=] AMREX_GPU_DEVICE (long ip) {
+        // This kernel supports multiple deposition orders (0, 1, 2, 3) at compile time.
+        // It calculates the momentum of ionized electrons based on equations (B8) and (B9)
+        // from the Massimo (2020) article and equation (14) from the C. Schroeder (2014) article.
+        // Additionally, it computes the energy of emitted electrons and assigns their properties
+        // (momentum, position, etc.) to newly created electrons in the plasma container.
+        amrex::AnyCTO(
+            amrex::TypeList<
+                amrex::CompileTimeOptions<0, 1, 2, 3>
+            >{}, {
+                Hipace::m_depos_order_xy
+            },
+            [&] (auto cto_func) {
+                amrex::ParallelForRNG(num_ions, cto_func);
+            },
+            [=] AMREX_GPU_DEVICE (long ip, const amrex::RandomEngine& engine,
+                                  auto depos_order_xy) {
 
             if(p_ion_mask[ip] != 0) {
+
+                // Avoid temp slice
+                const amrex::Real xp = x_prev[ip];
+                const amrex::Real yp = y_prev[ip];
+
+                if (amrex::ConstParticleIDWrapper(idcpup[ip]) < 0 ||
+                    !laser_bounds.contains(xp, yp)) return;
+
+                Complex A = 0;
+                Complex A_dx = 0;
+                Complex A_dzeta = 0;
+
+                doLaserGatherShapeN<depos_order_xy>(xp, yp, A, A_dx, A_dzeta, laser_arr,
+                    dx_inv, dy_inv, dzeta_inv, x_pos_offset, y_pos_offset);
+
+                amrex::Real ux = 0._rt;
+                amrex::Real uy = 0._rt;
+                amrex::Real uz = 0._rt;
+
+                if (linear_polarization) {
+                    // Get the level from which the electron was ionized.
+                    // The -1 is needed as this variable was incremented in the ionization kernel above.
+                    const int ion_lev_loc = ion_lev[ip]-1;
+                    const Complex Et = I * A * omega0 + A_dzeta * phys_const.c; // transverse component
+                    const Complex El = - A_dx * phys_const.c; // longitudinal component
+                    amrex::Real Ep = std::sqrt( amrex::abs(Et*Et) + amrex::abs(El*El) );
+                    Ep *= phys_const.m_e * phys_const.c / phys_const.q_e;
+                    Ep *= E0;
+                    amrex::Real delta = std::sqrt(Ep) * laser_dp_prefactor[ion_lev_loc];
+                    amrex::Real delta2 = delta * delta;
+                    amrex::Real width_p = amrex::abs(A) * delta * (1 - (3./4.) * delta2 - (3./2.) * delta2 + laser_dp_second_prefactor[ion_lev_loc] * delta2);
+                    amrex::Real p_pol = amrex::RandomNormal(0.0, width_p, engine);
+                    ux = p_pol; // linear polarization is assumed along x.
+                    uz = (amrex::abs(A * A) / 4._rt + p_pol * p_pol / 2._rt);
```

Comment:
```suggestion
                    amrex::Real width_p = amrex::abs(A) * delta * (1._rt - (3._rt/4._rt) * delta2
                        - (3._rt/2._rt) * delta2 + laser_dp_second_prefactor[ion_lev_loc] * delta2);
                    amrex::Real p_pol = amrex::RandomNormal(0.0, width_p, engine);
                    ux = p_pol; // linear polarization is assumed along x.
                    uz = (amrex::abs(A * A) * 0.25_rt + p_pol * p_pol * 0.5_rt);
```
Just one more small tweak to use _rt, make sure the line isn't too long, and replace division with multiplication for (maybe very very slightly) better performance.